library/ecjpake.c

```diff
@@ -286,6 +286,13 @@ static int ecjpake_zkp_read( const mbedtls_md_info_t *md_info,
      * Verification
      */
     MBEDTLS_MPI_CHK( ecjpake_hash( md_info, grp, pf, G, &V, X, id, &h ) );
+
+    if( mbedtls_mpi_cmp_int( &r,0 ) == 0 )
```

Comment:
Doesn't matter much, but for readability I'd have preferred this comparison to happen right after r is read in (i.e. move this block to line 284 instead)

Reply:
Suggestion: change line 276 above to `if( end < *p || (size_t)( end - *p ) < r_len || r_len == 0 )`. Then in all test cases that have `zero-length r` in the description, change the expected error from `MBEDTLS_ERR_ECP_INVALID_KEY` to `MBEDTLS_ERR_ECP_BAD_INPUT_DATA` (which is more consistent with the other cases of parsing errors).

Also, ideally we'd want the EC J-PAKE fix in its own commit (without a ChangeLog entry), and then the ECDSA fix (in `mul_shortcuts()`) in a second commit, with the ChangeLog entry.